File: src/main/java/medium/FindDuplicateNumber.java
package medium;

public class FindDuplicateNumber {
    public int solve(int[] nums) {
        int[] arr = new int[nums.length];
        for (int i : nums) {
            if (arr[i - 1] > 0) return i;
            arr[i - 1] = i;
        }
        return 0;
    }
}

File: src/main/php/Medium/FindDuplicateNumber.php
<?php

declare(strict_types=1);

namespace Src\Medium;

class FindDuplicateNumber
{
    public function solve(array $nums): int
    {
        $result = array_fill(0, count($nums), 0);
        for ($i = 0; $i < count($nums); $i++) {
            if ($result[$nums[$i] - 1] > 0) {
                return $nums[$i];
            }
            $result[$nums[$i] - 1]++;
        }

        return 0;
    }
}
